level_generator/level_generator.py

```diff
@@ -0,0 +1,178 @@
+    x, y = start_point
+    pref_directions =[]
+    if x > Tile.WIDTH//2:
+        pref_directions.append(Direction.LEFT.value)
+    else:
+        pref_directions += [Direction.RIGHT.value]
+    if y > Tile.HEIGHT//2:
+        pref_directions += [Direction.UP.value]
+    else:
+        pref_directions += [Direction.DOWN.value]
+
+    return pref_directions
+
+def draw_point(x:int, y:int, img:Image)->None:
+    """Create 1 point on a specific pixel on the image"""
+    img.putpixel((x, y), Colors.PATH)
+
+def my_random_direction(probability:dict) -> Direction:
+    """Return a random direction based on probabilities"""
+    vectors_probability = []
+    for i in range(probability[Direction.UP.value]):
+        vectors_probability.append(Direction.UP)
+    for i in range(probability[Direction.DOWN.value]):
+        vectors_probability.append(Direction.DOWN)
+    for i in range(probability[Direction.RIGHT.value]):
+        vectors_probability.append(Direction.RIGHT)
+    for i in range(probability[Direction.LEFT.value]):
+        vectors_probability.append(Direction.LEFT)
+    
+    return random.choice(vectors_probability)
+
+def move_to(rnd_direction:Direction, x:int, y:int) -> tuple:
+    """Change x, y to a new value based on the Direction"""
+    x += rnd_direction.value[0]
+    y += rnd_direction.value[1]
+    return x, y
+
+def end_points(start_point:tuple)->list[tuple]:
+    """Creates a list of coords of legal end points"""
+    if start_point[0] == 0:
+        return [(Tile.WIDTH-1, y) for y in range(Tile.HEIGHT)]
+    elif start_point[1] == 0:
+        return [(x, Tile.HEIGHT-1) for x in range(Tile.WIDTH)]
+    elif start_point[0] == Tile.WIDTH-1:
+        return [(0, y) for y in range(Tile.HEIGHT)]
+    else:
+        return [(x, 0) for x in range(Tile.WIDTH)]
+
+def legal_move(x:int, y:int):
+    return 0<=x<Tile.WIDTH and 0<=y<Tile.HEIGHT
+
+def generate_level(lvl:int, difficulty:int):
+    """Generate a level"""
+
+    probability = {
+        Direction.UP.value: 250,
+        Direction.DOWN.value: 250,
+        Direction.LEFT.value: 250,
+        Direction.RIGHT.value: 250
+    }
+
+    img = Image.new('RGB', (Tile.WIDTH, Tile.HEIGHT), color = (0, 0, 0))
+
+    #Vytvoří random start na random ose a pak vybere 1 start 
+    start_osa_1 = 0, random.randint(5, Tile.HEIGHT-5)
+    start_osa_2 = random.randint(5, Tile.WIDTH-5), 0
+    start_osa_3 = Tile.WIDTH-1, random.randint(5, Tile.HEIGHT-5)
+    start_osa_4 = random.randint(5, Tile.WIDTH-5), Tile.HEIGHT-1
+    #Vybraný start
+    chosen_start = random.choice([start_osa_1, start_osa_2, start_osa_3, start_osa_4])
+
+    # Define the border pixels
+    border_pixels = set(
+        [(x, 0) for x in range(Tile.WIDTH)] + 
+        [(x, Tile.HEIGHT-1) for x in range(Tile.WIDTH)] + 
+        [(0, y) for y in range(1, Tile.HEIGHT-1)] + 
+        [(Tile.WIDTH-1, y) for y in range(1, Tile.HEIGHT-1)]
+    )
+
+    #Start direction
+    first_direction_1 = first_direction(chosen_start)
```

Comment:
Proc tam je to _1 za first_direction? Promenna se muze jmenovat stejne, jak funkce. To samy u pref_directions a end_points.